src/App.tsx

```diff
@@ -2,59 +1,187 @@
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [showPostLoader, setShowPostLoader] = useState<boolean>(false);
+  const [showCommentLoader, setShowCommentLoader] = useState<boolean>(false);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isSidebarVisible, setIsSidebarVisible] = useState<number | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [hasPostError, setHasPostError] = useState<boolean>(false);
+  const [hasCommentError, setHasCommentError] = useState<boolean>(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(res => {
+        setUsers(res as User[]);
+      })
+      .catch(() => new Error('Failed to fetch users'));
+  }, []);
 
-              <Loader />
+  const fetchUserPosts = (user: User) => {
+    setShowPostLoader(true);
+    getUserPosts(user.id)
+      .then(res => {
+        setUserPosts(res as Post[]);
+        setHasPostError(false);
+      })
+      .catch(() => {
+        setHasPostError(true);
+      })
+      .finally(() => {
+        setShowPostLoader(false);
+      });
+  };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const fetchPostComments = (currentPost: Post) => {
+    setIsSidebarVisible(currentPost.id);
+    setShowCommentLoader(true);
+
+    setSelectedPost(currentPost);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    getUserPostComments(currentPost.id)
+      .then(res => {
+        setPostComments(res as Comment[]);
+        setHasCommentError(false);
+      })
+      .catch(() => {
+        setHasCommentError(true);
+      })
+      .finally(() => {
+        setShowCommentLoader(false);
+      });
+  };
+
+  const removeComment = (commentId: number) => {
+    deleteComment(commentId)
+      .then(() => {
+        setPostComments(prevPostComments =>
+          prevPostComments.filter(
+            (comment: Comment) => comment.id !== commentId,
+          ),
+        );
+      })
+      .catch(() => new Error('Failed to delete comment'));
+  };
+
+  const createComment = (newComment: CommentData) => {
+    setIsAddingComment(true);
+    addNewComment(newComment)
+      .then(res => {
+        setPostComments([...postComments, res as Comment]);
+        setIsAddingComment(false);
+      })
+      .catch(() => new Error('Failed to add comment'));
```

Comment:
You should handle the error in catch block. It's better to set some state variable in case of error and show error message to the user.